core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -648,7 +648,16 @@ impl ZksNamespace {
             .get_l1_batch_metadata(l1_batch_number)
             .await
             .map_err(|err| internal_error(METHOD_NAME, err))?
-            .map(|l1_batch_with_metadata| l1_batch_with_metadata.construct_pubdata().into());
+            .map(|l1_batch_with_metadata| {
+                let partial_pubdata = l1_batch_with_metadata.construct_pubdata();
+                // The encoding of empty pubdata results in a vector of zeroes. We should return
+                // `None in this case.
+                if partial_pubdata == vec![0u8; partial_pubdata.len()] {
```

Comment:
Could we consider using `iter().all()` to compare if all elements of `partial_pubdata` are zeroes?
This approach can potentially optimize performance by avoiding unnecessary allocation of a new vector of zeroes, particularly beneficial for larger vectors. The suggested change would look like:
```rust
if partial_pubdata.iter().all(|&x| x == 0) {
    Bytes::default()
} else {
    partial_pubdata.into()
}
```
Your thoughts on this optimization would be appreciated. Thank you!